src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts

```diff
@@ -475,26 +477,49 @@ export class SQLFuture extends Disposable implements FutureInternal {
 
 	private async processResultSets(batch: BatchSummary): Promise<void> {
 		try {
+			let queryRowsPromises: Promise<void>[] = [];
 			for (let resultSet of batch.resultSetSummaries) {
 				let rowCount = resultSet.rowCount > this.configuredMaxRows ? this.configuredMaxRows : resultSet.rowCount;
 				if (rowCount === this.configuredMaxRows) {
 					this.handleMessage(localize('sqlMaxRowsDisplayed', "Displaying Top {0} rows.", rowCount));
 				}
-				await this.sendResultSetAsIOPub(rowCount, resultSet);
+				queryRowsPromises.push(this.getAllQueryRows(rowCount, resultSet));
+			}
+			// We want to display table in the same order
+			let i = 0;
+			for (let resultSet of batch.resultSetSummaries) {
+				await queryRowsPromises[i];
+				this.sendResultSetAsIOPub(resultSet);
+				i++;
 			}
 		} catch (err) {
 			// TODO should we output this somewhere else?
 			this.logService.error(`Error outputting result sets from Notebook query: ${err}`);
 		}
 	}
 
-	private async sendResultSetAsIOPub(rowCount: number, resultSet: ResultSetSummary): Promise<void> {
-		let subsetResult: QueryExecuteSubsetResult;
+	private async getAllQueryRows(rowCount: number, resultSet: ResultSetSummary): Promise<void> {
+		let deferred: Deferred<void> = new Deferred<void>();
 		if (rowCount > 0) {
-			subsetResult = await this._queryRunner.getQueryRows(0, rowCount, resultSet.batchId, resultSet.id);
+			this._queryRunner.getQueryRows(0, rowCount, resultSet.batchId, resultSet.id).then((result) => {
+				this._querySubsetResultMap.set(resultSet.id, result);
+				deferred.resolve();
```

Comment:
Instead of having this map why not just create chained promises for each batch query that will call this.sendIOPubMessage(subsetResult, resultSet) after the query finishes executing? I like that a lot better than map lookups

Reply:
The problem with this is that we can't just output it immediately. We need to wait on the queries before that to return. Any ideas? I'd love to get rid of the map.

Reply:
Talked offline - we need to preserve order here so this is a fine approach